src/services/stringCompletions.ts

```diff
@@ -842,13 +843,16 @@ function getCompletionEntriesForNonRelativeModules(
 
     const result = createNameAndKindSet();
     const moduleResolution = getEmitModuleResolutionKind(compilerOptions);
+
+    const projectDir = compilerOptions.project || host.getCurrentDirectory();
+    const absolute = normalizePath(combinePaths(projectDir, baseUrl ?? getPathsBasePath(compilerOptions, host)));

```

Comment:
This is what I mean—all this is done for you in `getPathsBasePath`

```suggestion
    const absolute = getPathsBasePath(compilerOptions, host);
```

Reply:
I thought of this when trying to figure out what you meant. But the `absolute` variable is also used for the path completion relative to `baseUrl`, and the `getPathsBasePath` will return `undefined` when `compilerOptions.paths` is falsy, which breaks existing completion. I could also make it:

```ts
    if (baseUrl) {
        const projectDir = compilerOptions.project || host.getCurrentDirectory();
        const absolute = normalizePath(combinePaths(projectDir, baseUrl));
        getCompletionEntriesForDirectoryFragment(fragment, absolute, extensionOptions, host, /*moduleSpecifierIsRelative*/ false, /*exclude*/ undefined, result);
    }

    if (paths) {
        const absolute = getPathsBasePath(compilerOptions, host)!;
        addCompletionEntriesFromPaths(result, fragment, absolute, extensionOptions, host, paths);
    }
```

But I thought this doesn't make it more concise and keeps "the logic for making the absolute path" in one place. There is also this `compilerOptions.project` check that is different from the original code. But I guess that is not super important since I didn't see other language features considering that.

Reply:
`compilerOptions.project` is definitely incorrect in this existing code, as it’s the CLI-only `-p` option that points to a tsconfig file or directory 👀. It should always be `undefined` in the language service, as far as I know. That’s partly why I suggested using the utility function; I thought the old reference to `project` looked a little sketchy. I think the snippet you have above looks fine, though I think you could safely remove the existing reference to `compilerOptions.project`.